src/main/java/com/couchbase/lite/Database.java

```diff
@@ -1194,7 +1194,9 @@ public RevisionInternal getDocument(String docID, String revID, boolean withBody
 
     @InterfaceAudience.Private
     public RevisionInternal loadRevisionBody(RevisionInternal rev) throws CouchbaseLiteException {
-        return store.loadRevisionBody(rev);
+        synchronized(store) {
+            return store.loadRevisionBody(rev);
+        }
     }
```

Comment:
As we only synchronize the store on loadRevisionBody and getRevisionHistory method, I think we should put some comments here and explain why we are doing it on these 2 methods.